src/lib/update-alternatives/model/alternative.rb

```diff
@@ -95,6 +95,17 @@ def empty?
       false
     end
 
+    def choice(new_choice_path)
+      return unless choices.map(&:path).include?(new_choice_path)
+      @value = new_choice_path
+      @status = "manual"
+    end
+
+    def automatic_mode
+      @status = "auto"
+      @value = choices.sort { |a, b| b.priority.to_i <=> a.priority.to_i }.first.path
```

Comment:
you can use sort_by to make it easier -> `@value = choices.sort_by { |i| i.priority.to_i }.first.path